alerting/src/main/kotlin/org/opensearch/alerting/MonitorRunner.kt

```diff
@@ -127,16 +132,50 @@ abstract class MonitorRunner {
         monitorCtx: MonitorRunnerExecutionContext
     ): NotificationActionConfigs {
         var destination: Destination? = null
-        val channel: NotificationConfigInfo? = getNotificationConfigInfo(monitorCtx.client as NodeClient, action.destinationId)
+        var notificationPermissionException: Exception? = null
+
+        var channel: NotificationConfigInfo? = null
+        try {
+            channel = getNotificationConfigInfo(monitorCtx.client as NodeClient, action.destinationId)
+        } catch (e: OpenSearchSecurityException) {
+            notificationPermissionException = e
```

Comment:
I follow what we're doing here now but since the control flow has become a little more complex, am I understanding it correctly that:

* If we fail to get the channel because of security permissions and we also don't have permissions to destination **OR** there was no destination with that ID **then we throw the security exception for getting channel**
* If we are able to get the channel, the destination part is not executed so security permission issues are not a problem
* If there is no Notification config (but we have permissions to retrieve them) and then we fail to get the Destination because of security permissions **then we throw the Destination security permission**

Reply:
Yes that is correct.